src/app/components/DataLoader/DataLoader.jsx

```diff
@@ -1,10 +1,73 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import Actions from '@Actions';
+import { ajaxCall } from '@utils';
+import appConfig from '@appConfig';
 
 class DataLoader extends React.Component {
+  constructor(props) {
+    super(props);
+    const { location } = this.props;
+    this.pathInstructions = [
+      {
+        expression: /\/research\/collections\/shared-collection-catalog\/bib\/(b\d*)/,
+        pathType: 'bib',
+      },
+    ];
+    this.pathType = null;
+    this.routes = {
+      bib: {
+        apiRoute: () => `${appConfig.baseUrl}/api/bib?bibId=${this.matchData[1]}`,
+        actions: [Actions.updateBib],
+        errorMessage: 'Error attempting to make an ajax request to fetch a bib record from ResultsList',
+      },
+    };
+    this.reducePathExpressions = this.reducePathExpressions.bind(this);
+  }
+
+  componentDidMount() {
+    // Needs to add trackdiscovery actions
+    window.dataLoaderLocation = window.dataLoaderLocation || [];
+    window.dataLoaderLocation.push(this.props.location);
+    this.matchData = this.pathInstructions
+      .reduce(this.reducePathExpressions, null);
+    console.log('mounting pathType ', this.pathType)
+    if (this.routes[this.pathType]) {
+      const {
+        apiRoute,
+        actions,
+        errorMessage,
+      } = this.routes[this.pathType];
+      Actions.updateLoadingStatus(true);
+      console.log('calling ajax with ', apiRoute())
+      ajaxCall(apiRoute(),
+        (response) => {
+          actions.forEach(action => action(response.data));
+          Actions.updateLoadingStatus(false);
+        },
+        (error) => {
+          Actions.updateLoadingStatus(false);
+          console.error(
+            errorMessage,
+            error,
+          );
+        },
+      );
+    }
+  }
+
+  reducePathExpressions(acc, instruction) {
+    const { location } = this.props;
+    console.log('pathname', location.pathname)
+    const matchData = location.pathname.match(instruction.expression);
+    if (matchData) this.pathType = instruction.pathType;
+    return matchData || acc;
+  }
+
   render() {
-    return(
-      <div className="dataLoader">'Blah blah blah'</div>
-    )
+    return (
+      <div className="dataLoader" />
+    );
```

Comment:
The component itself makes sense and can help reduce API calls and make sure data is available before a component renders. But I think this can be improved to better explain what the purpose of this component is. Since this is at the very top application level and children component _depend_ on this, instead of this being an empty div sibling, it should be a wrapper around everything.

So here you can do 
```
render() {
  return (<>{this.props.children}</>);
}
```

And in `application.jsx`, 
```
<DataLoader location={this.context.router.location}>
  {React.cloneElement(this.props.children, this.state.data)}
</DataLoader>
```

It doesn't seem like it's necessary around the Header and Footer. I think this better defines the component and its usage.

Reply:
This doesn't really change anything, but semantically and reading it, I think it's more appropriate.

Reply:
Like this, implemented.